src/MonitorProfiler/MainProfiler/MainProfiler.cpp

```diff
@@ -45,3 +63,31 @@ STDMETHODIMP MainProfiler::LoadAsNotficationOnly(BOOL *pbNotificationOnly)
 
     return S_OK;
 }
+
+HRESULT MainProfiler::InitializeEnvironment()
+{
+    m_pEnvironment.reset(new (nothrow) ProfilerEnvironment(m_pCorProfilerInfo));
```

Comment:
std::make_shared?

Reply:
For some reason, I don't think this was compiling either, but maybe I'm misremembering. I'll try it again.